contrib/lib-autopilot.py

```diff
@@ -0,0 +1,289 @@
+        #self.__logger.addHandler(fh)
+        
+    def __load_nodes(self):
+        #FIXME: it is a real problem that we don't know how many nodes there could be
+        nodes = []
+        try:
+            self.__logger.info("Attempt RPC-call to download nodes from the lightning network")
+            while len(nodes) == 0: 
+                peers = self.__rpc_interface.listpeers()["peers"]
+                if len(peers) < 1:
+                    #FIXME: do we have seed nodes?
+                    self.__rpc_interface.connect("024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365")
+                    time.sleep(2)
+                nodes = self.__rpc_interface.listnodes()["nodes"]
+        except ValueError as e:
+            self.__logger.info("Node list could not be retrieved from the peers of the lightning network")
+            self.__logger.debug("RPC error: " + str(e))
+            return False
+        self.__logger.info("Number of nodes found: {}".format(len(nodes)))
+
+        for node in nodes:
+            self.G.add_node(node["nodeid"],**node)
+        return True
+    
+    def __load_edges(self):
+        """
+        tries to store channels of the lightning network to the networkx graph. the attributes of the
+        channel will be stored as edge attributes
+        """
+        if len(self.G.nodes)==0:
+            self.__add_logger("cannot download channels if nodes do not exist. Try downloading the node list...")
+            if self.__load_nodes() == False:
+                self.add_logger("Stop trying to download channels")
+                return False
+        
+        channels = {}
+        try: 
+            self.__logger.info("Attempt RPC-call to download channels from the lightning network")
+            channels = self.__rpc_interface.listchannels()["channels"]
+            self.__logger.info("Number of retrieved channels: {}".format(len(channels)))
+        except ValueError as e:
+            self.__logger.info("Channel list could not be retrieved from the peers of the lightning network")
+            self.__logger.debug("RPC error: " + str(e))
+            return False
+        
+        for channel in channels:
+            self.G.add_edge(channel["source"],channel["destination"],**channel)
+        
+        return True
+
+    def __generate_rich_nodes(self,k=3,percentile=0.5):
+        """
+            generates a set of nodes that are very well connected (such nodes with low degree will be prefered)
+        """
+        if k < 3:
+            k = 3
+        
+        if percentile < 0.0 or percentile > 1.0:
+            percentile = 0.5
+        
+        self.__logger.info("RICH_NODES: Try to generate {} candidates from the {}-percentile".format(k,percentile))
+        
+        rich_nodes = {}
+        network_capacity = 0
+        candidates = []
+        for n in self.G.nodes():
+            total_capacity = sum(self.G.get_edge_data(n,m)["satoshis"] for m in self.G.neighbors(n))
```

Comment:
We could also just annotate the nodes in the graph with their capacity when creating the graph. That'd save us some common computation later.

Reply:
The same would also be true for the node degree.

Reply:
yes, that would be great so I could expect that in the RPC-Interface? also it would be nice if satoshis would be renamed to capacity

Reply:
No, I don't think we should add non-primary fields, such as these derived values in the RPC, especially if they're this trivial to compute externally.

Reply:
not sure what to do with this. I continue to work on the "satoshi" field. I could build a bridge and might do so (especially when later calculating network flow becomes an option)